### ISSUE TITLE
Train accuracy high, low validation accuracy

### ISSUE THREAD

**Issue by GeorgeXia1828** [dmlc/mxnet#3378]
I am trying do the inception-bn-full on my own dataset. (near 1400w and 5190 classes).

python train_imagenet.py --network inception-bn-full --batch-size 100 --lr 0.05 --lr-factor 0.9 --gpus 2,3 --num-epoch 60 --data-dir /data5/rd/xiajizhong/parts_17kClasses/ --train-dataset small_rec/ --val-dataset small_rec_val/ --num-examples=14040000

and, when use the latest mxnet I get bad validation accuracy. 

> 2016-09-24 02:49:24,212 Node[0] Epoch[0] Batch [140400] Speed: 47.30 samples/sec    Train-accuracy=0.546400
> 2016-09-24 02:49:24,213 Node[0] Epoch[0] Batch [140400] Speed: 47.30 samples/sec    Train-top_k_accuracy_5=0.787400
> 2016-09-24 02:49:24,213 Node[0] Epoch[0] Batch [140400] Speed: 47.30 samples/sec    Train-top_k_accuracy_10=0.842400
> 2016-09-24 02:49:24,213 Node[0] Epoch[0] Batch [140400] Speed: 47.30 samples/sec    Train-top_k_accuracy_20=0.884800
> 2016-09-24 02:49:25,074 Node[0] Update[140401]: Change learning rate to 4.50000e-02
> 2016-09-24 02:49:31,389 Node[0] Epoch[0] Resetting Data Iterator
> 2016-09-24 02:49:31,390 Node[0] Epoch[0] Time cost=301141.343
> 2016-09-24 02:49:31,645 Node[0] Saved checkpoint to "model/inception_bn_full_Vdian-0001.params"
> 2016-09-24 05:15:15,472 Node[0] Epoch[0] Validation-accuracy=**0.142758**
> 2016-09-24 05:15:15,472 Node[0] Epoch[0] Validation-top_k_accuracy_5=0.280551
> 2016-09-24 05:15:15,472 Node[0] Epoch[0] Validation-top_k_accuracy_10=0.350004
> 2016-09-24 05:15:15,472 Node[0] Epoch[0] Validation-top_k_accuracy_20=0.418618

But, I was doing the experiment well in the old version mxnet, which the validation accuracy is also very high near top1=50%, after one epoch. So I am wondering why this happened? I am not very clear about different changed in the new version mxnet. Is it caused by the Pooling layer behavior changed? 

and on the old version, the result below,

> 2016-06-10 10:26:15,591 Node[0] Epoch[0] Batch [140350] Speed: 50.40 samples/sec    Train-accuracy=0.406846
> 2016-06-10 10:27:53,871 Node[0] Epoch[0] Batch [140400] Speed: 50.88 samples/sec    Train-accuracy=0.406867
> 2016-06-10 10:27:53,875 Node[0] Update[140401]: Change learning rate to 4.05000e-01
> 2016-06-10 10:27:59,645 Node[0] Epoch[0] Resetting Data Iterator
> 2016-06-10 10:27:59,647 Node[0] Epoch[0] Train-accuracy=0.406869
> 2016-06-10 10:27:59,647 Node[0] Epoch[0] Time cost=258948.773
> 2016-06-10 10:28:01,107 Node[0] Saved checkpoint to "model/vdian5190-0-0001.params"
> 2016-06-10 12:50:10,700 Node[0] Epoch[0] Validation-accuracy=**0.457665**

**Comment by George19890617** [comment]
I have read the source code of softmax_output-inl.h, but found the normalization is default set to be kNULL, so I think it's no difference in old and new? Is this true?

**Comment by winstywang** [comment]
If you do not meet InferShape Error, then it should not be pooling issue. The most probable cause is BatchNorm layer, especially if you use fix_gamma.

**Comment by George19890617** [comment]
@winstywang you mean the BatchNorm layer changed in new version mxnet? Because I trained and  got good result both train and val in old version? Can you explain more specific?

**Comment by winstywang** [comment]
@George19890617 Yes, you had better double check it.

**Comment by George19890617** [comment]
@winstywang thanks so much for you time. 
I am turn back to the mxnet 0.5.0 is OK. If you train successfully on the latest the mxnet, can you tell me more about your lr-schechulr? Actually, I set --lr 0.5 --lr-factor 0.9 --lr-factor-epoch 0.5 is OK in old one.

**Comment by winstywang** [comment]
Please check the latest pre-trained model here. https://github.com/dmlc/mxnet/pull/3506
